### PR TITLE
Add delay to unattended-upgrades after boot

### PR DESCRIPTION
**Pull request by nadeemshahzad**
Configuration Pull Request
---

Make sure that the following steps are done before merging:

  - [ ] A DevOps team member has approved the PR.
  - [ ] Are you adding any new default values that need to be overridden when this change goes live? If so:
    - [ ] Update the appropriate internal repo (be sure to update for all our environments)
    - [ ] If you are updating a secure value rather than an internal one, file a DEVOPS ticket with details.
    - [ ] Add an entry to the CHANGELOG.
  - [ ] If you are making a complicated change, have you performed the proper testing specified on the [Ops Ansible Testing Checklist](https://openedx.atlassian.net/wiki/display/EdxOps/Ops+Ansible+Testing+Checklist)?  Adding a new variable does not require the full list (although testing on a sandbox is a great idea to ensure it links with your downstream code changes).

### CODE REVIEW

File: playbooks/roles/security/templates/etc/systemd/system/apt-daily.timer.d/override.conf [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[Timer]
+OnBootSec=2h
```
Comment (feanil) [inline playbooks/roles/security/templates/etc/systemd/system/apt-daily.timer.d/override.conf]: Let's just set this to 1d so we don't have to worry about it while the AMI is being cut at all.  2h is above the usual build times but we could hit 2h if the build is really slow for some reason(slow connection for example).

**Review by feanil — COMMENTED** [review]
@nadeemshahzad  I think you also need to deal with the apt-daily-upgrade.timer right?

File: playbooks/roles/security/defaults/main.yml [hunk 2]
```diff
@@ -16,7 +16,7 @@
 #
 security_role_name: security
 # set to true to enable unattended upgrades nightly
-SECURITY_UNATTENDED_UPGRADES: false
+SECURITY_UNATTENDED_UPGRADES: true
```
Comment (jibsheet) [inline playbooks/roles/security/defaults/main.yml]: Do we want this true for every opensource user too or just for us.
Also, are we going to take this on our AMI instances?  I assumed this was only for machines that stay up for a long time (like say jenkins)

Comment (feanil) [reply playbooks/roles/security/defaults/main.yml]: Good point, we should still default this to false here.

I think we should take it on all our machines, with the override change, it should not interfere with any ansible runs and so it should be good to leave it running everywhere.

Comment (jibsheet) [reply playbooks/roles/security/defaults/main.yml]: We should probably sort out if we want the upgrades running on our AMI'd machines, but either way this should default back to false and we can make appropriate overrides.  The rest of this looks fine given we switch out to a day.

**Comment by nadeemshahzad** [comment]
Override configuration will add the delay of 24 hours to run ```apt-daily.service``` to update ache and ```apt-daily-upgrade.service``` to install security updates.
```
~# systemctl list-timers apt-daily.timer
NEXT                         LEFT     LAST PASSED UNIT            ACTIVATES
Tue 2018-02-20 23:58:21 UTC  23h left n/a  n/a    apt-daily.timer apt-daily.service
~# systemctl list-timers apt-daily-upgrade.timer
NEXT                         LEFT     LAST PASSED UNIT                    ACTIVATES
Wed 2018-02-21 00:14:09 UTC  24h left n/a  n/a    apt-daily-upgrade.timer apt-daily-upgrade.service
```

**Review by coryleeio — APPROVED** [review]

**Comment by feanil** [comment]
@nadeemshahzad do you have a sandbox running with this change?

**Comment by feanil** [comment]
Is this PR still relevant?  Seems like we haven't had this issue in a while?

**Comment by nadeemshahzad** [comment]
@feanil I think we can keep these changes for machines which we don't build with AMI and we can override` SECURITY_UNATTENDED_UPGRADES: true ` if we need to run for all other machines.
I have build sandbox with changes with override `SECURITY_UNATTENDED_UPGRADES: true` https://tools-edx-jenkins.edx.org/job/Sandboxes/job/CreateSandbox/14445/ .

**Pull request closed without merging**